tests/test_mc_simulator.py

```diff
@@ -610,3 +612,228 @@ def test_MantarrayMcSimulator__allows_status_bits_to_be_set_through_testing_queu
+        SERIAL_COMM_SIMPLE_COMMAND_PACKET_TYPE,
+        bytes([SERIAL_COMM_REBOOT_COMMAND_BYTE]),
+    )
+    simulator.write(test_reboot_command)
+    invoke_process_run_and_check_errors(simulator)
+
+    # test that reboot response packet is sent
+    expected_reboot_response = create_data_packet(
+        spied_get_cms_since_init.spy_return,
+        SERIAL_COMM_MAIN_MODULE_ID,
+        SERIAL_COMM_SIMPLE_COMMAND_PACKET_TYPE,
+        bytes(0),
+    )
+    reboot_response = simulator.read(size=len(expected_reboot_response))
+    assert reboot_response == expected_reboot_response
+
+    # test that handshake is ignored
+    test_handshake = create_data_packet(
+        dummy_timestamp,
+        SERIAL_COMM_MAIN_MODULE_ID,
+        SERIAL_COMM_HANDSHAKE_PACKET_TYPE,
+        bytes(0),
+    )
+    simulator.write(test_handshake)
+    invoke_process_run_and_check_errors(simulator)
+    response_during_reboot = simulator.read(size=HANDSHAKE_RESPONSE_SIZE_BYTES)
+    assert len(response_during_reboot) == 0
+
+    # test that status beacon is sent after reboot
+    invoke_process_run_and_check_errors(simulator)
+    expected_beacon = create_data_packet(
+        spied_get_cms_since_init.spy_return,
+        SERIAL_COMM_MAIN_MODULE_ID,
+        SERIAL_COMM_STATUS_BEACON_PACKET_TYPE,
+        DEFAULT_SIMULATOR_STATUS_CODE,
+    )
+    status_beacon = simulator.read(size=len(expected_beacon))
+    assert status_beacon == expected_beacon
+
+    # test that start time was reset
+    spied_reset.assert_called_once()
+
+
+def test_MantarrayMcSimulator__reset_status_code_after_rebooting(
+    mantarray_mc_simulator_no_beacon, mocker
+):
+    simulator = mantarray_mc_simulator_no_beacon["simulator"]
+    testing_queue = mantarray_mc_simulator_no_beacon["testing_queue"]
+
+    reboot_duration = 5
+    mocker.patch.object(
+        mc_simulator,
+        "_get_secs_since_reboot_command",
+        autospec=True,
+        return_value=reboot_duration,
+    )
+
+    # set status code to known value
+    expected_status_code_bits = bytes([1, 7, 3, 8])
+    test_command = {
+        "command": "set_status_code_bits",
+        "status_code_bits": expected_status_code_bits,
+    }
+    put_object_into_queue_and_raise_error_if_eventually_still_empty(
+        test_command, testing_queue
+    )
+    invoke_process_run_and_check_errors(simulator)
+
+    # send reboot command
+    dummy_timestamp = 0
+    test_reboot_command = create_data_packet(
+        dummy_timestamp,
+        SERIAL_COMM_MAIN_MODULE_ID,
+        SERIAL_COMM_SIMPLE_COMMAND_PACKET_TYPE,
+        bytes([SERIAL_COMM_REBOOT_COMMAND_BYTE]),
+    )
+    simulator.write(test_reboot_command)
+    invoke_process_run_and_check_errors(simulator)
+
+    # remove reboot response packet
+    invoke_process_run_and_check_errors(simulator)
+    expected_reboot_response = create_data_packet(
+        0,
+        SERIAL_COMM_MAIN_MODULE_ID,
+        SERIAL_COMM_SIMPLE_COMMAND_PACKET_TYPE,
+        bytes(0),
```

Comment:
would the status here be expected to be `expected_status_code_bits` (since that's the status of the Mantarray when the reboot command is received by the Mantarray)? I thought that was the way the Mantarray worked, but maybe I'm misunderstanding how the test is working or how the Mantarray works

Reply:
This is the "command acknowledged"  message it returns before rebooting. After completing reboot it sends a status beacon with the status bits